js/main.js

```diff
@@ -9,75 +9,197 @@ xhr.addEventListener('load', function () {
+        $parentCard.style.backgroundSize = 'cover';
+      } else if (event.target.className === 'far fa-circle 3') {
+        clearDots();
+        event.target.className = 'fas fa-circle 3';
+        $parentCard.style.background = 'linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.4)), url(' + 'https://ddragon.leagueoflegends.com/cdn/img/champion/loading/' + $key + '_' + skinNumArray[skinNumArray.length - 2] + '.jpg' + ')';
+        $parentCard.style.backgroundSize = 'cover';
+      } else if (event.target.className === 'far fa-circle 4') {
+        clearDots();
+        event.target.className = 'fas fa-circle 4';
+        $parentCard.style.background = 'linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.4)), url(' + 'https://ddragon.leagueoflegends.com/cdn/img/champion/loading/' + $key + '_' + skinNumArray[skinNumArray.length - 3] + '.jpg' + ')';
+        $parentCard.style.backgroundSize = 'cover';
+      }
+    });
+    xhr3.send();
+  }
+  if (event.target.className === 'read-more-link') {
+    $parentCard = event.target.closest('.card');
+    $key = $parentCard.querySelector('h2').textContent;
+    var $lore = $parentCard.querySelector('p');
+    var xhr4 = new XMLHttpRequest();
+    xhr4.open('GET', 'http://ddragon.leagueoflegends.com/cdn/11.22.1/data/en_US/champion/' + $key + '.json');
+    xhr4.responseType = 'json';
+    xhr4.addEventListener('load', function () {
+      $lore.textContent = xhr4.response.data[$key].lore;
+    });
+    xhr4.send();
   }
 }
 
-var $allCards = document.querySelectorAll('.column-half');
 function showRole(role) {
   for (var i = 0; i < $allCards.length; i++) {
     if (!$allCards[i].getAttribute('roles').includes(role)) {
       $allCards[i].style.display = 'none';
-    } else { $allCards[i].style.display = 'flex'; }
+    } else {
+      $allCards[i].style.display = 'flex';
+    }
   }
 }
 
 var $form = document.querySelector('form');
 $form.addEventListener('submit', handleSubmit);
 function handleSubmit(event) {
   event.preventDefault();
+  showSearchedChampions($searchBar.value);
 }
 
 var $searchBar = document.querySelector('input');
 $searchBar.addEventListener('input', handleInput);
+$searchBar.addEventListener('click', handleInput);
 function handleInput(event) {
   showSearchedChampions($searchBar.value);
 }
 
 function showSearchedChampions(name) {
   var currentTag = document.querySelector('.selected');
   for (var i = 0; i < $allCards.length; i++) {
-    if ($allCards[i].childNodes[0].childNodes[2].textContent.toLowerCase().includes(name.toLowerCase()) && $allCards[i].getAttribute('roles').toLowerCase().includes(currentTag.textContent.toLowerCase())) {
+    if ($allCards[i].childNodes[0].childNodes[0].textContent.toLowerCase().includes(name.toLowerCase()) && $allCards[i].getAttribute('roles').toLowerCase().includes(currentTag.textContent.toLowerCase())) {
```

Comment:
You can use `querySelector` on `$allCards[i]` to avoid all this `childNode` usage